src/App.jsx

```diff
@@ -14,45 +23,132 @@ export const goodsFromServer = [
-      </button>
-
-      <button
-        type="button"
-        className="button is-warning is-light"
-      >
-        Reverse
-      </button>
-
-      <button
-        type="button"
-        className="button is-danger is-light"
-      >
-        Reset
-      </button>
-    </div>
+function updateCurrentGoods(goods, sortType, isReversed) {
+  let visibleGoods = [...goods];
+
+  const sortedGoods = visibleGoods.sort((good1, good2) => {
+    switch (sortType) {
+      case SORT_BY_ALPHABET:
+        return good1.localeCompare(good2);
+      case SORT_BY_LENGTH:
+        return good1.length - good2.length;
+      default:
+        return 0;
+    }
+  });
+
+  if (isReversed) {
+    visibleGoods = visibleGoods.reverse();
+  }
+
+  return isReversed ? visibleGoods : sortedGoods;
+}
+
+function createSortText(sortType, isReversed) {
+  let result;
+
+  switch (sortType) {
+    case SORT_BY_ALPHABET:
+      result = NAME_SORT_BY_ALPHABET;
+      break;
+    case SORT_BY_LENGTH:
+      result = NAME_SORT_BY_LENGTH;
+      break;
+    default:
+      result = NAME_SORT_ENOUGH;
+      break;
+  }
+
+  if (isReversed) {
+    result += ' and the list is reversed';
+  }
+
+  return result;
+}
+
+export const App = () => {
+  const [sortType, setSortType] = useState('');
+  const [isReversed, setIsReversed] = useState(false);
 
-    <ul>
-      <li data-cy="Good">Dumplings</li>
-      <li data-cy="Good">Carrot</li>
-      <li data-cy="Good">Eggs</li>
-      <li data-cy="Good">Ice cream</li>
-      <li data-cy="Good">Apple</li>
-      <li data-cy="Good">...</li>
-    </ul>
-  </div>
-);
+  const currentGoods = updateCurrentGoods(
+    goodsFromServer,
+    sortType,
+    isReversed,
+  );
+
+  const reset = () => {
+    setSortType('');
+    setIsReversed(false);
+  };
+
+  const isReset = sortType || isReversed;
+
+  return (
+    <div className="section content">
+      <div className="buttons">
+        <button
+          type="button"
+          className={
+            cn(
+              'button is-info',
+              { 'is-light': sortType !== SORT_BY_ALPHABET },
+            )
+          }
+          onClick={() => {
+            setSortType(SORT_BY_ALPHABET);
+          }}
+        >
+          Sort alphabetically
+        </button>
+
+        <button
+          type="button"
+          className={
+            cn(
+              'button is-success',
+              { 'is-light': sortType !== SORT_BY_LENGTH },
+            )
+          }
+          onClick={() => {
+            setSortType(SORT_BY_LENGTH);
+          }}
+        >
+          Sort by length
+        </button>
+
+        <button
+          type="button"
+          className={
+            cn(
+              'button is-warning',
+              { 'is-light': !isReversed === true },
+            )
+          }
+          onClick={() => {
+            setIsReversed(!isReversed);
+          }}
+        >
+          Reverse
+        </button>
+
+        {
+          isReset
```

Comment:
Instead of using a ternary operator to conditionally render the reset button, you can use short-circuit evaluation like 'isReset && (...button code...)'